homeassistant/components/nest/device_info.py

```diff
@@ -50,9 +56,9 @@ def device_name(self):
         return self.device_model
 
     @property
-    def device_model(self):
+    def device_model(self) -> str:
         """Return device model information."""
         # The API intentionally returns minimal information about specific
         # devices, instead relying on traits, but we can infer a generic model
         # name based on the type
-        return DEVICE_TYPE_MAP.get(self._device.type)
+        return DEVICE_TYPE_MAP.get(self._device.type, "Unknown")
```

Comment:
If we don't know the model we don't set it at all in the device info or set it to `None`.

Reply:
Should DeviceInfo's model field have `| None` or am I misunderstanding the typing?

https://github.com/home-assistant/core/blob/7590cb2861f55d72c5f7b459e78d4f9f01b7a50c/homeassistant/helpers/entity.py#L172

Reply:
Currently we handle the case when `model` is `None` so we should update the typing. Although in the long run we should consider cleaning up and making it more strict.

https://github.com/home-assistant/core/blob/7590cb2861f55d72c5f7b459e78d4f9f01b7a50c/homeassistant/helpers/device_registry.py#L60